### PR TITLE
Add Functional Comment(s) Feature

### PR DESCRIPTION
**Pull request by isong2**
**Description:**
This PR adds the comments feature to the repository.

    1. Comments Feature --> Adds an MVP implementation of the 'comment' feature for an activity post. There is a button associated with activity posts that opens a modal containing all the comments specific to a post, which change-makers can read from. Additionally, change-makers can also now leave comments on posts.

    2. Future Necessities --> Unit tests, text wrapping, character limits, automatic scrolling down on adding a comment, displaying user profile pictures along with their comments, automatic filtering system, pagination of retrieval.
    
Below is a screenshot of the current comments feature.

Screen Shot 2023-03-31 at 12 09 11 AM
    
**Resolves:** #14 #19

### CODE REVIEW

**Review by tylowe-labs — COMMENTED** [review]
Few questions & concerns:

- Does the comment modal scroll down and show the comment added by a user? This was not happening when I was working on the branch earlier
- Does the profile image loader work because that also was not working when I was working on this
- Add a screenshot of the comment UI and it working as part of the PR
- Depending on what you need to do, add subtasks to the Kanban so we can keep track of the parts of this that are not finished yet and they don't get lost

**Review by tylowe-labs — APPROVED** [review]
Approving this for now; however, @Jareltey @Bwillia2 really should examine this PR closely + figure out how you are going to sort out the polishing tasks mentioned previously.

File: libs/client/shell/src/lib/activityfeed/activityposts/comments/modal-comments.component.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+  this.profilePic$ = this.user.session.selectors.changeMaker$.pipe(
+    map(changeMaker => changeMaker?.profilePicFilePath || null)
+  );
+
+  this.name$ = this.user.session.selectors.changeMaker$.pipe(
+    map(changeMaker => `${changeMaker?.firstName || ''} ${changeMaker?.lastName || ''}` || '')
+  );
+
+  // In Progress - Need to figure out how to show new comments as they are created
```
Comment (Bwillia2) [inline libs/client/shell/src/lib/activityfeed/activityposts/comments/modal-comments.component.ts:52]: Dead code can be deleted now that comments are updating correctly

**Review by Bwillia2 — APPROVED** [review]
I've left commented out code that's obsolete now that I will remove in my next commit.

**Comment by isong2** [comment]
The above commits were for the purposing of rebasing and merging.

**Pull request merged**